### ISSUE TITLE
gdb7.sdf.csv has a different number of samples than gdb7.sdf

### ISSUE THREAD

**Issue by lilleswing** [deepchem/deepchem#1048]
```python
max_num_atoms = 23
input_file = "gdb7.sdf"
tasks = ["u0_atom"]
dataframes = []
raw_df = next(load_csv_files([input_file + ".csv"], shard_size=None))
clean_mols = False
suppl = Chem.SDMolSupplier(str(input_file), clean_mols, False, False)
print(len(suppl))
print(len(raw_df))
```

qm7.mat has the same as the sdf file

@miaecle

**Comment by miaecle** [comment]
This is weird, I did not make these files but they should be consistent.
On the other hand I don't think we are still using `gdb7.sdf ` and `gdb7.sdf.csv` by default, should we only point everyone to `qm7.mat`?

**Comment by joegomes** [comment]
I authored these files.  The sdf was created by taking the coordinated and atomic numbers supplied in `qm7.mat`, creating .xyz files, then using babel to convert to sdf.  The csv was created by pulling the raw labels from `qm7.mat`.  There were some issues with `qm7.sdf` that we ran into early on, some of the supplied coordinates were incorrect (CH4 was one I recall).  When the coordinates were re-optimized at a similar level of QM theory, the coulomb matrix of the corrected structures closely resembled the supplied features.  After that, I think we decided to just use the raw features and labels from `qm7.mat` for QM7 models since the correct coordinates the features were produced from for a few structures were unavailable.  I would consider `qm7.sdf` and `qm7.sdf.csv` deprecated.

The different lengths for the csv and sdf files are likely the reason why there is a discrepancy in the test error for QM7 models.  It should be possible to recreate the QM7 structures and labels  from QM9, for the purpose of testing other featurization methods, AFAIK those coordinates are correct.
